http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -754,6 +782,180 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
       .build
     F.delay(ComputeManager.authenticate(clientSecretCredential, azureProfile))
   }
+
+  private def getCommonFields(name: String,
+                              description: String,
+                              app: App
+  ): bio.terra.workspace.model.ControlledResourceCommonFields = {
+    val commonFieldsBase = new bio.terra.workspace.model.ControlledResourceCommonFields()
+      .name(name)
+      .description(description)
+      .managedBy(bio.terra.workspace.model.ManagedBy.APPLICATION)
+      .cloningInstructions(CloningInstructionsEnum.NOTHING)
+    app.samResourceId.accessScope match {
+      case Some(AppAccessScope.WorkspaceShared) =>
+        commonFieldsBase.accessScope(bio.terra.workspace.model.AccessScope.SHARED_ACCESS)
+      case _ =>
+        commonFieldsBase
+          .accessScope(bio.terra.workspace.model.AccessScope.PRIVATE_ACCESS)
+          .privateResourceUser(
+            new bio.terra.workspace.model.PrivateResourceUser()
+              .userName(app.auditInfo.creator.value)
+              .privateResourceIamRole(bio.terra.workspace.model.ControlledResourceIamRole.WRITER)
+          )
+    }
+  }
+
+  private[util] def maybeCreateWsmIdentityAndDatabase(app: App,
+                                                      workspaceId: WorkspaceId,
+                                                      landingZoneResources: LandingZoneResources,
+                                                      namespace: KubernetesNamespace
+  )(implicit
+    ev: Ask[F, AppContext]
+  ): F[Option[ServiceAccountName]] = {
+    val databaseConfigEnabled = app.appType match {
+      case AppType.Wds => config.wdsAppConfig.databaseEnabled
+      case _           => false
+    }
+    val landingZoneSupportsDatabase = landingZoneResources.postgresName.isDefined
+    if (databaseConfigEnabled && landingZoneSupportsDatabase) {
+      for {
+        ctx <- ev.ask
+        _ <- logger.info(ctx.loggingCtx)(
+          s"Creating WSM identity for app ${app.appName.value} in cloud workspace ${workspaceId.value}"
+        )
+
+        // Build WSM client
+        auth <- samDao.getLeoAuthToken
+        token <- auth.credentials match {
+          case org.http4s.Credentials.Token(_, token) => F.pure(token)
+          case _ => F.raiseError(new RuntimeException("Could not obtain Leo auth token"))
+        }
+        wsmApi = wsmClientProvider.getControlledAzureResourceApi(token)
+
+        // Build create managed identity request.
+        // Use the k8s namespace for the name. Note dashes aren't allowed.
+        identityName = s"id${namespace.name.value.split('-').head}"
```

Comment:
Naming is hard -- the identity/db names can't be too long, and can't have dashes. Leo namespace names are 6 random alphanumeric characters.